src/connector/src/parser/mod.rs

```diff
@@ -493,6 +495,142 @@ pub enum SpecificParserConfig {
     Bytes,
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct ParserClientConfigList {
+    username: Option<String>,
+    password: Option<String>,
+}
+
+impl From<&HashMap<String, String>> for ParserClientConfigList {
+    fn from(props: &HashMap<String, String>) -> Self {
+        const SCHEMA_REGISTRY_USERNAME: &str = "schema.registry.username";
+        const SCHEMA_REGISTRY_PASSWORD: &str = "schema.registry.password";
+
+        ParserClientConfigList {
+            username: props.get(SCHEMA_REGISTRY_USERNAME).cloned(),
+            password: props.get(SCHEMA_REGISTRY_PASSWORD).cloned(),
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct CsvParserConfigList {
+    pub delimiter: u8,
+    pub has_header: bool,
+}
+
+#[derive(Default)]
+pub struct AvroParserConfigList {
+    pub use_schema_registry: bool,
+    pub row_schema_location: String,
+    pub aws_auth_props: Option<AwsAuthProps>,
+    pub topic: String,
+    pub client_config: ParserClientConfigList,
+    pub primary_key: String,
+}
+
+#[derive(Default)]
+pub struct ProtoParserConfigList {
+    pub message_name: String,
+    pub use_schema_registry: bool,
+    pub row_schema_location: String,
+    pub aws_auth_props: Option<AwsAuthProps>,
+    pub topic: String,
+    pub client_config: ParserClientConfigList,
+}
+
+#[derive(Default)]
+pub struct DebeziumAvroParserConfigList {
```

Comment:
We should not concat Avro and debezium into one config. 
It is expected to do the same with separate configs.

Reply:
This is for Debezium avro only